notebooks/stac-load-S2-deafrica.py

```diff
@@ -0,0 +1,154 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py:percent
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.13.6
+#   kernelspec:
+#     display_name: 'Python 3.8.12 64-bit (''stac'': conda)'
+#     language: python
+#     name: python3
+# ---
+
+# %% [markdown]
+# # Access Sentinel 2 Analysis Ready Data from Digital Earth Africa
+#
+# [![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/opendatacube/odc-stac/develop?labpath=notebooks%2Fstac-load-S2-deafrica.ipynb)
+#
+# https://explorer.digitalearth.africa/products/s2_l2a
+
+# %% [markdown]
+# ## Import required packages
+
+# %%
+import rasterio
+from pystac_client import Client
+
+from odc.stac import stac_load
+
+# %% [markdown]
+# ## Set configuration
+#
+# The configuration dictionary is determined from the product's definition, availble at https://explorer.digitalearth.africa/products/s2_l2a#definition-doc
+#
+# All assets except SLC have the same configuration. SLC uses `uint8` rather than `uint16`.
+#
+# In the configuration, we also supply the aliases for each band. This means we can load data by band name rather than band number.
+
+# %%
+config = {
+    "s2_l2a": {
+        "assets": {
+            "*": {
+                "data_type": "uint16",
+                "nodata": 0, 
+                "unit": "1",
+            },
+            "SLC": {
+                "data_type": "uint8",
+                "nodata": 0, 
+                "unit": "1",
+            },
+        },
+        "aliases": {
+            "costal_aerosol": "B01",
+            "blue": "B02",
+            "green": "B03",
+            "red": "B04",
+            "red_edge_1": "B05",
+            "red_edge_2": "B06",
+            "red_edge_3": "B07",
+            "nir": "B08",
+            "nir_narrow": "B08A",
+            "water_vapour": "B09",
+            "swir_1": "B11",
+            "swir_2": "B12",
+            "mask": "SLC",
+            "aerosol_optical_thickness": "AOT",
+            "scene_average_water_vapour": "WVP",
+        }
+    }
+}
+
+# %% [markdown]
+# ## Connect to the Digital Earth Africa STAC catalog
+
+# %%
+# Open the stac catalogue
+deafrica_stac_address = 'https://explorer.digitalearth.africa/stac'
+catalog = Client.open(deafrica_stac_address)
+
+# %% [markdown]
+# ## Find STAC Items to Load
+
+# %%
+# Construct a bounding box to search over
+# [xmin, ymin, xmax, ymax] in latitude and longitude
+bbox = [37.76, 12.49, 37.77, 12.50]
+
+# Construct a time range to search over
+start_date = '2020-09-01'
+end_date = '2020-12-01'
+timerange = f'{start_date}/{end_date}'
+
+# Choose the product/s to load
+products = ['s2_l2a']
+
+# %%
+# Identify all data matching the above:
+query = catalog.search(
+    bbox=bbox,
+    collections=products,
+    datetime=timerange
```

Comment:
I'm not sure whether indirection via `timerange` adds value here.
1. It does not match the name used by stac client/api
2. But using the same name is also no good, as `datetime` is an important module that you don't want to alias

I think just using `datetime=f'{start_date}/{end_date}'` would be more readable. You can include extra explainer that brings attention to `start/stop` syntax used by stac client for time ranges if you think this might be missed without defining `timerange=` variable.

Reply:
Done